js/modules/k6/http/http_request.go

```diff
@@ -257,6 +258,15 @@ func (h *HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.Stat
 		buf := state.BPool.Get()
 		buf.Reset()
 		defer state.BPool.Put(buf)
+
+		if res.Header.Get("Content-Encoding") == "deflate" {
+			var err error
+			res.Body, err = zlib.NewReader(res.Body)
+			if err != nil {
+				resErr = err
```

Comment:
This is going to mess up the `io.Copy()` below if it fails, because it's gonna replace `res.Body` with a broken/nil reader (not sure which, it's probably undefined).

Add another `if resErr == nil && res != nil` clause above to do this in instead, then you can straight up do `res.Body, resErr = …`.